lib/p2p/NodeList.ts

```diff
@@ -132,7 +132,11 @@ class NodeList extends EventEmitter {
    * @return true if the specified node exists and the event was added, false otherwise
    */
   public addReputationEvent = async (nodePubKey: string, event: ReputationEvent): Promise<boolean> => {
-    const node = this.nodes.get(nodePubKey);
+    let node;
+    if (this.isBanned(nodePubKey)) {
+      node = this.bannedNodes.get(nodePubKey);
+    } else node = this.nodes.get(nodePubKey);
```

Comment:
I'd prefer keeping the `const` here and doing an in-line if like `const node = this.isBanned(nodePubKey) : this.bannedNodes.get(nodePubKey) | this.nodes.get(nodePubKey);`